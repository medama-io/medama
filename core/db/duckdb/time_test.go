package duckdb_test

import (
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

func TestGetWebsiteTimeSummary(t *testing.T) {
	_, require, ctx, client := UseDatabaseFixture(t, SIMPLE_FIXTURE)

	testCases := getBaseTestCases(MEDIUM_HOSTNAME)

	for _, tc := range testCases {
		t.Run(tc.Name, func(t *testing.T) {
			summary, err := client.GetWebsiteTimeSummary(ctx, tc.Filters)
			require.NoError(err)

			snap := NewSnapRecords(summary)
			snaps.MatchSnapshot(t, snap.Snapshot())
		})
	}
}

func TestGetWebsiteTime(t *testing.T) {
	_, require, ctx, client := UseDatabaseFixture(t, SIMPLE_FIXTURE)

	testCases := getBaseTestCases(MEDIUM_HOSTNAME)

	for _, tc := range testCases {
		t.Run(tc.Name, func(t *testing.T) {
			times, err := client.GetWebsiteTime(ctx, tc.Filters)
			require.NoError(err)

			snap := NewSnapRecords(times)
			snaps.MatchSnapshot(t, snap.Snapshot())
		})
	}
}
