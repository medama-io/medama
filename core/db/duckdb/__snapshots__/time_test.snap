
[TestGetWebsiteTime/Base - 1]
RECORDS:
&{StatsTimeSummary:{Pathname:/about Duration:5002 DurationPercentage:0.3326} DurationUpperQuartile:7496 DurationLowerQuartile:2493 Pageviews:0 Visitors:166895}
&{StatsTimeSummary:{Pathname:/ Duration:5008 DurationPercentage:0.3336} DurationUpperQuartile:7492 DurationLowerQuartile:2490 Pageviews:0 Visitors:166610}
&{StatsTimeSummary:{Pathname:/contact Duration:4998 DurationPercentage:0.3337} DurationUpperQuartile:7498 DurationLowerQuartile:2509 Pageviews:0 Visitors:166353}

---

[TestGetWebsiteTime/Browser - 1]
RECORDS:
&{StatsTimeSummary:{Pathname:/about Duration:5000 DurationPercentage:0.3336} DurationUpperQuartile:7489 DurationLowerQuartile:2473 Pageviews:0 Visitors:55789}
&{StatsTimeSummary:{Pathname:/ Duration:4983 DurationPercentage:0.3327} DurationUpperQuartile:7476 DurationLowerQuartile:2458 Pageviews:0 Visitors:55260}
&{StatsTimeSummary:{Pathname:/contact Duration:4956 DurationPercentage:0.3337} DurationUpperQuartile:7465 DurationLowerQuartile:2500 Pageviews:0 Visitors:55121}

---

[TestGetWebsiteTime/Country - 1]
RECORDS:
&{StatsTimeSummary:{Pathname:/ Duration:5041 DurationPercentage:0.3346} DurationUpperQuartile:7486 DurationLowerQuartile:2511 Pageviews:0 Visitors:18558}
&{StatsTimeSummary:{Pathname:/about Duration:4975 DurationPercentage:0.33} DurationUpperQuartile:7459 DurationLowerQuartile:2446 Pageviews:0 Visitors:18499}
&{StatsTimeSummary:{Pathname:/contact Duration:4958 DurationPercentage:0.3354} DurationUpperQuartile:7461 DurationLowerQuartile:2544 Pageviews:0 Visitors:18311}

---

[TestGetWebsiteTime/Device - 1]
RECORDS:
&{StatsTimeSummary:{Pathname:/ Duration:5107 DurationPercentage:0.3384} DurationUpperQuartile:7535 DurationLowerQuartile:2559 Pageviews:0 Visitors:6305}
&{StatsTimeSummary:{Pathname:/about Duration:4981 DurationPercentage:0.3274} DurationUpperQuartile:7422 DurationLowerQuartile:2374 Pageviews:0 Visitors:6164}
&{StatsTimeSummary:{Pathname:/contact Duration:4894 DurationPercentage:0.3342} DurationUpperQuartile:7460 DurationLowerQuartile:2545 Pageviews:0 Visitors:6124}

---

[TestGetWebsiteTime/Empty - 1]
RECORDS:

---

[TestGetWebsiteTime/Language - 1]
RECORDS:
&{StatsTimeSummary:{Pathname:/ Duration:5145 DurationPercentage:0.3397} DurationUpperQuartile:7523 DurationLowerQuartile:2607 Pageviews:0 Visitors:3230}
&{StatsTimeSummary:{Pathname:/about Duration:4936 DurationPercentage:0.3288} DurationUpperQuartile:7388 DurationLowerQuartile:2390 Pageviews:0 Visitors:3075}
&{StatsTimeSummary:{Pathname:/contact Duration:4874 DurationPercentage:0.3315} DurationUpperQuartile:7434 DurationLowerQuartile:2672 Pageviews:0 Visitors:3073}

---

[TestGetWebsiteTime/OS - 1]
RECORDS:
&{StatsTimeSummary:{Pathname:/ Duration:5098 DurationPercentage:0.349} DurationUpperQuartile:7503 DurationLowerQuartile:2572 Pageviews:0 Visitors:1073}
&{StatsTimeSummary:{Pathname:/about Duration:4845 DurationPercentage:0.3289} DurationUpperQuartile:7325 DurationLowerQuartile:2388 Pageviews:0 Visitors:1068}
&{StatsTimeSummary:{Pathname:/contact Duration:4924 DurationPercentage:0.3221} DurationUpperQuartile:7435 DurationLowerQuartile:2797 Pageviews:0 Visitors:944}

---

[TestGetWebsiteTime/Pathname - 1]
RECORDS:
&{StatsTimeSummary:{Pathname:/ Duration:5098 DurationPercentage:1} DurationUpperQuartile:7503 DurationLowerQuartile:2572 Pageviews:0 Visitors:1073}

---

[TestGetWebsiteTime/Referrer - 1]
RECORDS:
&{StatsTimeSummary:{Pathname:/ Duration:4728 DurationPercentage:1} DurationUpperQuartile:7404 DurationLowerQuartile:2293 Pageviews:0 Visitors:359}

---

[TestGetWebsiteTime/UTMCampaign - 1]
RECORDS:
&{StatsTimeSummary:{Pathname:/ Duration:4732 DurationPercentage:1} DurationUpperQuartile:7502 DurationLowerQuartile:2291 Pageviews:0 Visitors:110}

---

[TestGetWebsiteTime/UTMMedium - 1]
RECORDS:
&{StatsTimeSummary:{Pathname:/ Duration:6547 DurationPercentage:1} DurationUpperQuartile:8733 DurationLowerQuartile:2558 Pageviews:0 Visitors:36}

---

[TestGetWebsiteTime/UTMSource - 1]
RECORDS:
&{StatsTimeSummary:{Pathname:/ Duration:3389 DurationPercentage:1} DurationUpperQuartile:6698 DurationLowerQuartile:2220 Pageviews:0 Visitors:16}

---

[TestGetWebsiteTimeSummary/Base - 1]
RECORDS:
&{Pathname:/contact Duration:4998 DurationPercentage:0.3337}
&{Pathname:/ Duration:5008 DurationPercentage:0.3336}
&{Pathname:/about Duration:5002 DurationPercentage:0.3326}

---

[TestGetWebsiteTimeSummary/Browser - 1]
RECORDS:
&{Pathname:/contact Duration:4956 DurationPercentage:0.3337}
&{Pathname:/about Duration:5000 DurationPercentage:0.3336}
&{Pathname:/ Duration:4983 DurationPercentage:0.3327}

---

[TestGetWebsiteTimeSummary/Country - 1]
RECORDS:
&{Pathname:/contact Duration:4958 DurationPercentage:0.3354}
&{Pathname:/ Duration:5041 DurationPercentage:0.3346}
&{Pathname:/about Duration:4975 DurationPercentage:0.33}

---

[TestGetWebsiteTimeSummary/Device - 1]
RECORDS:
&{Pathname:/ Duration:5107 DurationPercentage:0.3384}
&{Pathname:/contact Duration:4894 DurationPercentage:0.3342}
&{Pathname:/about Duration:4981 DurationPercentage:0.3274}

---

[TestGetWebsiteTimeSummary/Empty - 1]
RECORDS:

---

[TestGetWebsiteTimeSummary/Language - 1]
RECORDS:
&{Pathname:/ Duration:5145 DurationPercentage:0.3397}
&{Pathname:/contact Duration:4874 DurationPercentage:0.3315}
&{Pathname:/about Duration:4936 DurationPercentage:0.3288}

---

[TestGetWebsiteTimeSummary/OS - 1]
RECORDS:
&{Pathname:/ Duration:5098 DurationPercentage:0.349}
&{Pathname:/about Duration:4845 DurationPercentage:0.3289}
&{Pathname:/contact Duration:4924 DurationPercentage:0.3221}

---

[TestGetWebsiteTimeSummary/Pathname - 1]
RECORDS:
&{Pathname:/ Duration:5098 DurationPercentage:1}

---

[TestGetWebsiteTimeSummary/Referrer - 1]
RECORDS:
&{Pathname:/ Duration:4728 DurationPercentage:1}

---

[TestGetWebsiteTimeSummary/UTMCampaign - 1]
RECORDS:
&{Pathname:/ Duration:4732 DurationPercentage:1}

---

[TestGetWebsiteTimeSummary/UTMMedium - 1]
RECORDS:
&{Pathname:/ Duration:6547 DurationPercentage:1}

---

[TestGetWebsiteTimeSummary/UTMSource - 1]
RECORDS:
&{Pathname:/ Duration:3389 DurationPercentage:1}

---
